regression/polus-bleed-through-estimation-plugin/src/bench.py

```diff
@@ -0,0 +1,269 @@
+    coefficients_before = numpy.zeros(shape=(len(files), len(files)), dtype=numpy.float32)
+    coefficients_after = numpy.zeros(shape=(len(files), len(files)), dtype=numpy.float32)
+
+    if not (before_path.exists() and after_path.exists()):
+        processes: dict[tuple[int, int], Future[tuple[float, float]]] = dict()
+        with ProcessPoolExecutor(max_workers=utils.NUM_THREADS) as executor:
+            for i, image_path_1 in enumerate(files):
+                for j, image_path_2 in enumerate(files[i + 1:], start=i + 1):
+                    processes[(i, j)] = executor.submit(
+                        _pearson_thread,
+                        image_path_1,
+                        image_path_2,
+                        selected_tiles,
+                        corrected_dir,
+                    )
+
+            logger.info(f'calculating pearson coefficients...')
+            process_values = list(processes.values())
+            done, not_done = wait(process_values, 0)
+            while len(not_done) > 0:
+                logger.info(f'Percent complete: {100 * len(done) / len(process_values):6.3f}%')
+                for process in done:
+                    process.result()
+                done, not_done = wait(process_values, 15)
+            logger.info(f'Percent complete: {100 * len(done) / len(process_values):6.3f}%')
+
+        for (i, j), future in processes.items():
+            before, after = future.result()
+            coefficients_before[i, j] = coefficients_before[j, i] = before
+            coefficients_after[i, j] = coefficients_before[j, i] = after
+
+        with open(before_path, 'wb') as outfile:
+            pickle.dump(coefficients_before, outfile)
+        with open(after_path, 'wb') as outfile:
+            pickle.dump(coefficients_after, outfile)
+
+    else:
+        with open(before_path, 'rb') as infile:
+            coefficients_before = pickle.load(infile)
+        with open(after_path, 'rb') as infile:
+            coefficients_after = pickle.load(infile)
+
+    draw_heatmap(coefficients_before, coefficients_after, heatmaps_dir, heatmaps_name, replicate)
+
+    return
+
+
+def draw_heatmap(
+        before: numpy.ndarray,
+        after: numpy.ndarray,
+        heatmaps_dir: Path,
+        heatmaps_name: str,
+        replicate: int,
+):
+    heat = numpy.zeros_like(before, dtype=numpy.float32)
+    heat[numpy.tril_indices_from(heat)] = after[numpy.tril_indices_from(after)]
+    heat[numpy.triu_indices_from(heat)] = before[numpy.triu_indices_from(before)]
+    heat = heat.T
+
+    mask = numpy.zeros_like(heat, dtype=bool)
+    mask[numpy.diag_indices_from(mask)] = True
+
+    with seaborn.axes_style('white'):
+        _ = seaborn.heatmap(
+            heat, mask=mask, vmin=-1, vmax=1, cmap='vlag', linewidth=0.5,
+            annot=True, fmt='.3f', annot_kws={'size': 8},
+        )
+        matplotlib.pylab.title(f'Replicate {replicate}')
```

Comment:
why are you using pylab? If I'm not mistaken pyplot is preferred for non-interactive usage.

Reply:
I copied most of this code from examples and documentation. I'm also using both seaborn and matplotlib. Ideally I'd just use seaborn. I just decided that the plots looked good enough and it wasn't worth the time to make those changes.